cmd/builder/command.go

```diff
@@ -41,7 +51,19 @@ func (c *Command) Interactive() (err error) {
 	return
 }
 
+// Exec will send the command to shell execution.
+func (c *Command) Exec() (outStr string, err error) {
+	outStr, err = shell.Exec(c.command, c.args...)
+	return
+}
+
 // String returns a string representation of the command.
 func (c *Command) String() string {
 	return strings.Trim(fmt.Sprintf("%s %s", c.command, strings.Join(c.args, " ")), " ")
 }
+
+// LookPath returns if the command exists
+func (c *Command) LookPath() (err error) {
+	_, err = exec.LookPath(c.command)
```

Comment:
Scratching my head thinking on how  to test this. Well at least we could have a dummy test with an unknown executable and verify the error is returned.

To mock the LookPath return then I am not sure (non-error return); have that in a safe way... maybe checking for `go` command? (assuming tests are executed in an environment that MUST have `go` installed, right)

Reply:
@fabriciojs yes, we need to think better in an easy and fashionable way to mock both `exec.LookPath` and `exec.Command`. Make them easily replaceable for testing purposes.